microsoft/devhome
# Customization: Add general system and virtualization feature settings

## Summary of the pull request
Adds virtualization feature management and general system settings initial implementation.
 
## References and relevant issues
#2920 #3249 #2871

## Detailed description of the pull request / Additional comments
- Updates `ManagementInfrastructureHelper.cs` for enhanced Windows feature management, including a refactored method for checking feature availability with modifications to return more detailed feature information.
- Introduced a `RestartHelper.cs` class with a method to restart the computer immediately, to avoid disruption I did not point existing callers from Environments to this method since they have additional telemetry for this space. I can file an issue to track consolidation here.
- Updated `WindowsIdentityHelper.cs` with a new method to check for whether a user has the capability to acquire administrative privileges, not just whether the current process is running elevated.
- Added `WindowsOptionalFeatures.cs` and `WindowsOptionalFeatureState.cs` to represent Windows optional features, including a new static class for feature names and descriptions and a class to represent the state of an optional feature, specifically for UI.
- Introduced `ModifyLongPathsSetting.cs` and `ModifyWindowsOptionalFeatures.cs` for enabling/disabling Windows settings and optional features through PowerShell scripts which require elevation. This allows the full script content to be visible to the user when accepting or rejecting the UAC prompt providing extra transparency, this is done in the same way as environments feature management. There are limitations here in that if scripts are blocked on the machine we're unable to make this changes, but until we finish security review on the elevated server, this is the best we've got.
- Expanded Windows Customization with new views and dialogs for managing general system settings and virtualization features, including `GeneralSystemView`, `ModifyFeaturesDialog`, and `VirtualizationFeatureManagementPage`, alongside updates to existing views to accommodate these new features.

## Validation steps performed
- Test toggling a single and multiple features and confirm that the correct action is taken.
- Revert changes and confirm that the InfoBar warning instructing to restart is cleared
- Navigate away and come back to see that the InfoBar warning is still present.
- Confirm that restart buttons correctly restart the machine.
- Test cancelation with multiple features and ensure that the UI accurately reflects the state of the machine.
- Confirm that registry values are set appropriately for long path settings.

## PR checklist
- [X ] Closes #2920, #3249, #2871
- [ X] Telemetry [compliance tasks](https://aka.ms/devhome-telemetry) completed for added/updated events
